docs/contributing/source/ways-can-contribute/vanshdeep-notes.md

```diff
@@ -0,0 +1,163 @@
+
+```bash
+
+$ echo $GOPATH
+/home/user/Documents/go
+$ go -h
+Go is a tool for managing Go source code.
+
+Usage:
+
+	go command [arguments]
+
+The commands are:
+
+	build       compile packages and dependencies
+	clean       remove object files
+
+...
+
+```
+
+- We can drop `GOPATH` in `PATH` so that installed Golang packages are directly available in the shell. To do that add the following to your `.bashrc`
+
+```bash
+export PATH=${PATH}:${GOPATH}/bin
+```
+
+- Now we can install Glide which is vendor package manger for Golang,
+
+```bash
+$ go get github.com/Masterminds/glide
+
+...
+...
+
+$ glide -h
+NAME:
+   glide - Vendor Package Management for your Go projects.
+
+   Each project should have a 'glide.yaml' file in the project directory. Files
+   look something like this:
+
+...
+```
+
+- Now we install Tendermint from source,
+
+```bash
+$ mkdir -p $GOPATH/src/github.com/tendermint && cd $_
+$ git clone https://github.com/tendermint/tendermint
+...
+$ cd tendermint && glide install
+$ go install ./cmd/tendermint
+```
+
+- If the above commands were executed successfully then Tendermint is installed at `$GOPATH/bin`. To ensure Tendermint's installed fine execute the following command,
+
+```bash
+$ tendermint -h
+Tendermint Core (BFT Consensus) in Go
+
+Usage:
+  tendermint [command]
+
+Available Commands:
+  gen_validator               Generate new validator keypair
+  help                        Help about any command
+  init                        Initialize Tendermint
+...
+
+```
+
+### Running Tendermint
+- We can initialize and run tendermint as follows,
+```bash
+$ tendermint init
+...
+
+$ tendermint node --consensus.create_empty_blocks=false
+```
+The argument `--consensus.create_empty_blocks=false` specifies that Tendermint should not commit empty blocks.
+
+
+- To reset all the data stored in Tendermint execute the following command,
+
+```bash
+$ tendermint unsafe_reset_all
+```
+
+## Installing BigchainDB
+
+To install BigchainDB from source (for dev), clone the repo and execute the following command, (it is better that you create a virtual env for this)
+
+```bash
+$ git clone https://github.com/bigchaindb/bigchaindb.git
+...
+$ git checkout tendermint
+$ pip install -e .[dev]  #  or  pip install -e '.[dev]'  # for zsh
+```
+
+
+## Running tests
+
+To execute tests when developing a feature or fixing a bug one could use the following command,
+
+```bash
+$ pytest -v --database-backend=localmongodb
+```
+
+NOTE: MongoDB and Tendermint should be running as discussed above.
+
+One could mark a specific test and execute the same by appending `-m my_mark` to the above command.
+
+Although the above should prove sufficient in most cases but in case tests are failing on Travis CI then the following command can be used to possibly replicate the failure locally,
+
+```bash
+$ docker-compose -f docker-compose.travis.yml run --rm --no-deps bdb pytest -v --cov=bigchaindb
```

Comment:
This has been changed by @muawiakh PR #2130

Reply:
Oh dear, I was worried about that. @kansi can you check what the latest way to use docker-compose is, here?

Reply:
@ttmc the docker compose files have been merged into a single file so no need to specify the file, i.e.

```bash
$ docker-compose run --rm --no-deps bdb pytest -v --cov=bigchaindb
```